DOCSP-16309 Creating databases and collections using the MongoDB playgrounds (#49)
mongodb/docs-mongodb-vscode: source/playground-databases.txt

.. _vsce-playground-databases:

================================

> Creating Databases & Collections

Author: Done.

> Creating Databases & Collections

Author: Done.

================================
.. default-domain:: mongodb

.. contents:: On this page
   :local:
   :backlinks: none
   :depth: 1
   :class: singlecol

.. include:: /includes/fact-vsce-preview.rst

You can create 
:manual:`databases and collections </core/databases-and-collections/>` 
using a MongoDB Playground. 

Prerequisites
-------------

If you have not done so already, you must complete the following 
prerequisites before you can create a time series collection with a 
MongoDB Playground:

- :ref:`Create a connection to a MongoDB deployment 
  <vsce-connect-task>`.
- :ref:`Activate the connection to the MongoDB deployment 
  <vsce-activate-connection>`.

Create a Database and Collection
--------------------------------

Once you connect to your deployment using MongoDB for VS code, use the left 
navigation to: 

1. Select an active connection and click the :icon-fa5:`plus` icon that 
   appears.
   
#. A MongoDB playground automatically opens with a template form to create 
   a database and a regular collection or a 
   :ref:`time series collection <vsce-time-series>`. 

> #. Fill in the provided fields with the name of your database and collection.

Reviewer: I was a little confused at first which "provided" fields we meant here. Maybe we can explicitly say "Fill in the `const database` and `const collection` fields..."

Author: Done.

#. Uncomment the form of the collection you wish to create and fill in the 

>    provided fields accordingly.

Reviewer: [suggestion]

These steps feel slightly out of order / confusing to me since it kind of seems like we're telling users to fill in their collection info twice. Maybe we do something like:

```
# Uncomment the form of the collection you wish to create.
# Fill in the collection fields according to your collection's specifications.
```

Author: Done.

.. _vsce-playground-databases:

==============================

> Create Databases & Collections

Author: Done.

using a MongoDB Playground. 

Prerequisites
-------------

If you have not done so already, you must complete the following 
prerequisites before you can create a database or collection with a 
MongoDB Playground:

- :ref:`Create a connection to a MongoDB deployment 
  <vsce-connect-task>`.
- :ref:`Activate the connection to the MongoDB deployment 
  <vsce-activate-connection>`.

Create a Database and Collection
--------------------------------

Once you connect to your deployment using MongoDB for VS code, use the left 
navigation to: 

1. Select an active connection and click the :icon-fa5:`plus` icon that 
   appears.
   
#. A MongoDB playground automatically opens with a template form to create 
   a database and a regular collection or a 
   :ref:`time series collection <vsce-time-series>`. 

#. Fill in ``const database`` and ``const collection`` with names of your 
   choice for your database and collection.

#. Uncomment the form of the collection you wish to create.

#. Fill in the collection fields according to your collection's specifications.

#. To run the playground, click the :guilabel:`Play Button` at the top right
   of the VS code navigation bar.

> After running the playground, the left navigation will update with new database 

Reviewer: Minor, but we can use present tense here.

```suggestion
After running the playground, the left navigation updates with new database 
```

Author: Done.

listed under the active connection it was created. When the new database is 

> expanded the new collection can be found there.

Reviewer: Let's make this last sentence active voice. Maybe something like:

"You can find the newly created collection by expanding the new database."

Author: Done.

.. seealso::

   - To add documents to the collection, see 
     :ref:`vsce-create-doc-playground`.  

   - To learn more about creating time series collections using MongoDB 
     playgrounds, see :ref:`vsce-time-series`

Example
-------

This example creates a database named ``grades`` and a regular collection 

> named ``testscores``

Reviewer: ```suggestion
named ``testscores``.
```

Author: Done.

To use this example, start with a template from your MongoDB 

> Playgrounds. 

Reviewer: ```suggestion
Playgrounds:
```

.. code-block:: javascript

   const database = 'grades';
   const collection = 'testscores';

   use(database);

   db.createCollection(collection);

In the example:

- ``const database`` declares the name ``grades`` for the database.
  
- ``const collection`` declares the name ``testscores`` for the collection. 

- ``use(database)`` creates the new ``grades`` database.

> - ``db.createCollection(collection)`` creates the collection ``testscores``.

Reviewer: Maybe we should specify:

... creates the collection `testscores` inside of the `grades` database.

Author: Done.

When you press the :guilabel:`Play Button`, MongoDB for VS Code splits your 

> Playground and outputs the following result in the Playground Results.json 

Reviewer: Maybe we can simplify this a bit and say say: "... MongoDB for VS Code shows the following result in the `Playground Results.json` pane to confirm the creation of the database and collection: "

Author: Done.

pane to confirm the creation of the database and collection.

.. code-block:: javascript

   {
      "ok": 1
   }

The ``grades`` database and ``testscores`` collection also appear in your 

> left navigation. 

Reviewer: ```suggestion
left navigation:
```

Author: Done.
